### PR TITLE
config for automatic release to jenkins-ci.org from travis

### PR DESCRIPTION
**Pull request by mrfatstrat**
I haven't been able to test the entire setup in one go but have verified each part individually. I needed a newer version of maven to do the release locally (upgraded from 3.0.4 to 3.3.9). Also, the user diabol_jenkins has not been tested to do the actual release, the user works ok to login to jenkins-ci.org with but the user would need access to github to since the maven-release-plugin is tagging and and pushing to the repo and maybe something else is needed, eg. to be preent as a contributer in the pom file?  It's a bit difficult to test the complete release process. I started the PR for some code review and discussions while we test it propperly and I don'r recommend merging before we are sure it works.

### CODE REVIEW

**Comment by mrfatstrat** [comment]
On the 0.9.11 release verified that diabol_jenkins was auhorized to make the release. Now only thing left to verify is that the release-plugin can run in travis and that travis can push the tag to github

File: settings.xml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  <pluginGroups>
+    <pluginGroup>org.jenkins-ci.tools</pluginGroup>
+  </pluginGroups>
+
+  <profiles>
+    <!-- Give access to Jenkins plugins -->
+    <profile>
+      <id>jenkins</id>
+      <activation>
+        <activeByDefault>true</activeByDefault> <!-- change this to false, if you don't like to have it on per default -->
```
Comment (tommysdk) [inline settings.xml]: Let's remove the comment at the end of the line

File: settings.xml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    <mirror>
+      <id>repo.jenkins-ci.org</id>
+      <url>http://repo.jenkins-ci.org/public/</url>
+      <mirrorOf>m.g.o-public</mirrorOf>
+    </mirror>
+  </mirrors>
+
+  <servers>
+    <server>
+      <id>maven.jenkins-ci.org</id> <!-- For parent 1.397 or newer; before this use id java.net-m2-repository -->
```
Comment (tommysdk) [inline settings.xml]: We can remove this comment as well imo

**Comment by tommysdk** [comment]
So how would this work in practice? What would you need to do to perform an actual release in Travis and how should you specify the desired release version?

**Comment by tommysdk** [comment]
Noticed that there is a spelling error in the commit message ("autoamtic"), can you amend the message?

**Comment by mrfatstrat** [comment]
@tommysdk  I'll try to fix the commit comment and remove the comments in setting.xml.  

It will work like this:  Each merge to master will trigger the release after a successful build. The version will be the current version in the pom file (minus SNAPSHOT) and after the version will be bumped and checked back in with SNAPSHOT ready for next release. If we want to bump the major och minor number we need to do it manually in a normal commit. Update of the wiki is not done so this remains as a manual step. If we don't want a release for every merge to master we might be able to trigger the release on a specific commit comment or similar.

**Comment by mrfatstrat** [comment]
been thinking a bit more about it and I think I prefer to manually triggering the release and I think it would be nice to do it with a tag. So that when a tag (with a specific format) is created the release kicks off. Problem is that the release-plugin also creates the release-tag so we would get duplicate tags for each release, we would either live with that or delete one of them afterward. We could also generate the release notes from the git log and add it to github thru the api, this way we could just reference the github releases page from the wiki and not have release notes there.

**Comment by tommysdk** [comment]
@mrfatstrat I also think it makes sense not to create a release for every merge as some merges contain non-feature commits, e.g. code cleanup, documentation updates etc which it would be unnecessary to create a release for. I think it is fine in this case to perform the release manually, but in an automated way (I assume that is still the goal).

I'm all in favor of having the release notes on GitHub if we can automate the process. It should be fairly simple to look at what issues have been commited and marked as resolved since the last release and include those for the new version.

I'm not completely following what you mean with duplicate tags. Could you describe what steps are necessary to perform a release at the moment?

**Comment by mrfatstrat** [comment]
@tommysdk ok let's go with that strategy, i.e. automatic release triggered manually by som kind of action in the repo (e.g. a tag or a commit). I'll do some more investigations around this to find a good approach.

FYI:
A release is done with these steps:

1) release:prepare : sets versions in the pom file, create backup files for rollback, and create a release-tag in the repo, bumps the version to the new dev-version in the pom-file and commitøs it back to the repo. creates a release.properties that is used in the perform step.
2) release:perform: checks out from tag, builds, tests and uploads the artifact to artifactory

As I understand you can't skip the tagging since the perform-step depends on it. So if we create a tag let's say "release-0.1.12" that triggers the release step, we end up with two tags, "release-0-1-12" and "delivery-pipeline-plugin-0.1.12" (default, but the format can be specified), no big deal maybe but it get's a bit confusing in the repo.

**Comment by tommysdk** [comment]
@mrfatstrat OK, then I understand. I don't think it's a problem having two tags, since we can just consider the extra one a label.

**Comment by KostyaSha** [comment]
Probably you would like to try https://github.com/jucies/releases ?

**Comment by tommysdk** [comment]
@KostyaSha Thank you for the pointer! It looks really promising!

@mrfatstrat Do you think you could take a look at it?

**Comment by tommysdk** [comment]
We should probably also create the Docker images automatically and push to Docker hub while automating this.

**Pull request merged**